nodebuilder/share/constructors.go

```diff
@@ -36,6 +36,24 @@ func discovery(cfg Config) func(routing.ContentRouting, host.Host) *disc.Discove
 	}
 }
 
+// ensurePeersLifecycle controls the lifecycle for discovering full nodes.
+// This constructor is in place of the peer manager which generally controls the
+// EnsurePeers lifecycle.
+func ensurePeersLifecycle(lc fx.Lifecycle, discovery *disc.Discovery) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	lc.Append(fx.Hook{
+		OnStart: func(context.Context) error {
+			go discovery.EnsurePeers(ctx)
```

Comment:
I don't really like the fact that we call it here, but don't have a better idea atm